src/agent/src/windows_api_facade.cpp

```diff
@@ -0,0 +1,22 @@
+#include <windows_api_facade.hpp>
+
+#define WIN32_LEAN_AND_MEAN
+#include <windows.h>
+
+namespace windows_api_facade
+{
+    void* WindowsApiFacade::OpenSCM(unsigned int desiredAccess)
+    {
+        return ::OpenSCManager(nullptr, nullptr, static_cast<DWORD>(desiredAccess));
+    }
+
+    void* WindowsApiFacade::OpenSvc(void* sHandle, std::string sName, unsigned int desiredAccess)
+    {
+        return ::OpenService(static_cast<SC_HANDLE>(sHandle), sName.c_str(), static_cast<DWORD>(desiredAccess));
+    }
+
+    bool WindowsApiFacade::DeleteSvc(void* sHandle)
+    {
+        return ::DeleteService(static_cast<SC_HANDLE>(sHandle));
```

Comment:
Can we use the full names without abbreviations? ie: `OpenSCManager, OpenService, DeleteService`.

Reply:
I didn't manage to do that, when I use the same name calls the windows api function. I think it's because inside the api the name is defined like this
```
#ifdef UNICODE
#define OpenSCManager  OpenSCManagerW
#else
#define OpenSCManager  OpenSCManagerA
#endif // !UNICODE
```

Reply:
:thinking: it shouldn't be a problem since these are member functions inside a separate namespace... we can have a look if you think is worth it!